# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1233270786 18000
#      Thu Jan 29 18:13:06 2009 -0500
# Node ID 4385bb503467d6cbd834378dd4023b1f5052b858
# Parent  ee70d438a5b41d7242103f2b70f19a04418bdb9c
pt-stmt.cc (tree_statemnt::eval): check in_function_of_script_body to decide whether to call octave_call_stack::set_statement

diff --git a/src/ChangeLog b/src/ChangeLog
--- a/src/ChangeLog
+++ b/src/ChangeLog
@@ -1,10 +1,14 @@
 2009-01-29  John W. Eaton  <jwe@octave.org>
 
+	* pt-stmt.cc (tree_statement::eval): Check
+	in_function_or_script_body instead of symbol_table::at_top_level ()
+	to decide whether to call octave_call_stack::set_statement.
+
 	* graphics.cc (properties::set_defaults): Also set
 	horizontalalignment, verticalalignmnt, and rotation properties for
 	new label objects here.
 
 	* ov-base-scalar.cc (octave_base_scalar<ST>::is_true (void) const):
 	Error if scalar is NaN.
 	* ov-base-mat.cc (octave_base_matrix<MT>::is_true (void) const):
 	Likewise, if any element of matrix is NaN.
diff --git a/src/pt-stmt.cc b/src/pt-stmt.cc
--- a/src/pt-stmt.cc
+++ b/src/pt-stmt.cc
@@ -84,17 +84,17 @@ tree_statement::eval (bool silent, int n
 		      bool in_function_or_script_body)
 {
   octave_value_list retval;
 
   bool pf = silent ? false : print_flag;
 
   if (cmd || expr)
     {
-      if (! symbol_table::at_top_level ())
+      if (in_function_or_script_body)
 	octave_call_stack::set_statement (this);
 
       maybe_echo_code (in_function_or_script_body);
 
       try
 	{
 	  if (cmd)
 	    cmd->eval ();
